packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -362,7 +369,7 @@ IOSink? _openSink(String? output) {
   if (output == 'stdout') {
     sink = stdout;
   } else {
-    file = File(output);
+    file = File(basePath + output);
```

Comment:
Paths shouldn't be concatenated like this, but with `path.join`.

(Applies throughout the file.)